packages/core/core/src/UncommittedAsset.js

```diff
@@ -182,8 +188,32 @@ export default class UncommittedAsset {
     this.clearAST();
   }
 
-  getMapBuffer(): Promise<?Buffer> {
-    return Promise.resolve(this.mapBuffer);
+  async loadExistingSourcemap(): Promise<?SourceMap> {
```

Comment:
Ideally this would run before any transformations so we always have the sourcemaps in the asset regardless of transformations calling getMap.

However we can't really do this as we don't actually know when a file can have a sourcemap or not...

Reply:
Just figured out I'll need to add a list of extensions that we want to automatically load sourcemaps from as currently it is way too likely that a transformer calls this on an asset that isn't a valid sourcemap target like the test that is failing. Besides this a transformer might also call this when it's halfway through a pipeline when it's probably no longer relevant.